src/data/sym/card.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import algebra.big_operators.basic
+import data.sym.sym2
+
+/-!
+# Stars and bars
+
+In this file, we prove the case `n = 2` of stars and bars.
+
+## Informal statement
+
+If we have `n` objects to put in `k` boxes, we can do so in exactly `(n + k - 1).choose n` ways.
+
+## Formal statement
+
+We can identify the `k` boxes with the elements of a fintype `α` of card `k`. Then placing `n`
+elements in those boxes corresponds to choosing how many of each element of `α` appear in a multiset
+of card `n`. `sym α n` being the subtype of `multiset α` of multisets of card `n`, writing stars
+and bars using types gives
+```lean
+lemma stars_and_bars {α : Type*} [fintype α] (n : ℕ) :
+  card (sym α n) = (card α + n - 1).choose (card α) := sorry
+```
+
+## TODO
+
+Prove the general case of stars and bars.
+
+## Tags
+
+stars and bars
+-/
+
+open finset fintype
+
+namespace sym2
+variables {α : Type*} [decidable_eq α]
+
+/-- The `off_diag` of `s : finset α` is sent on a finset of `sym2 α` of card `s.card.choose 2`. -/
+lemma card_image_diag (s : finset α) :
+  (s.diag.image quotient.mk).card = s.card :=
+begin
+  rw [card_image_of_inj_on, diag_card],
+  rintro ⟨x₀, x₁⟩ hx _ _ h,
+  cases quotient.eq.1 h,
+  { refl },
+  { simp only [mem_coe, mem_diag] at hx,
+    rw hx.2 }
+end
+
+/-- The `off_diag` of `s : finset α` is sent on a finset of `sym2 α` of card `s.off_diag.card / 2`.
+This is because every element `⟦(x, y)⟧` of `sym2 α` not on the diagonal comes from exactly two
+pairs : `(x, y)` and `(y, x)`. -/
+lemma card_image_off_diag (s : finset α) :
+  (s.off_diag.image quotient.mk).card = s.off_diag.card / 2 :=
+begin
```

Comment:
Can you split out the following lemma?
```lean
lemma two_mul_card_image_off_diag (s : finset α) :
  2 * (s.off_diag.image quotient.mk).card = s.off_diag.card :=
begin
```
Using a `2 * ` to prove a `/ 2` is easy, but using a `/ 2` to prove a `2 * ` is harder.